File: Telerik/SecondControl/03.SolveExpression.cs
using System;
using System.Text.RegularExpressions;

class SolveExpression
{
    static void Main()
    {
        string expression = Console.ReadLine();
        expression = Regex.Replace(expression, "\\s+", "");
        long value = CalculateExpression(expression);
        Console.WriteLine(value);
    }

    static long CalculateExpression(string expression)
    {
        if (!expression.Contains("("))
        {
            string pattern2 = @"^(\d+)([+*-])(\d+)$";
            Match match2 = Regex.Match(expression, pattern2);
            long num1 = long.Parse(match2.Groups[1].Value);
            long num2 = long.Parse(match2.Groups[3].Value);
            string operand2 = match2.Groups[2].Value;
            long value = 0;
            switch (operand2)
            {
                case "+":
                    value = num1 + num2;
                    break;
                case "-":
                    value = num1 - num2;
                    break;
                case "*":
                    value = num1 * num2;
                    break;
            }
            return value;
        }
        string pattern = @"^(\d+)([+*-])\(([\d()*+-]+)\)$";
        Match match = Regex.Match(expression, pattern);
        long number = long.Parse(match.Groups[1].Value);
        string operand = match.Groups[2].Value;
        string subexpression = match.Groups[3].Value;
        switch (operand)
        {
            case "+":
                return (number + CalculateExpression(subexpression));
                break;
            case "-":
                return (number - CalculateExpression(subexpression));
                break;
            case "*":
                return (number * CalculateExpression(subexpression));
                break;
        }
        return 0;
    }
}

File: Telerik/SecondControl/03.SolveExpression.java
package Telerik;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class SolveExpression {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String expression = scan.nextLine();
        expression = expression.replaceAll("\\s+", "");
        Long value = CalculateExpression(expression);
        System.out.println(value);
    }

    static Long CalculateExpression(String expression) {
        if (!expression.contains("(")) {
            String pattern2 = "^(\\d+)([+*-])(\\d+)$";
            Pattern ptr2 = Pattern.compile(pattern2);
            Matcher match2 = ptr2.matcher(expression);
            if (match2.find()) {
                Long num1 = Long.parseLong(match2.group(1));
                Long num2 = Long.parseLong(match2.group(3));
                String operand2 = match2.group(2);
                Long value = 0l;
                switch (operand2) {
                    case "+":
                        value = num1 + num2;
                        break;
                    case "-":
                        value = num1 - num2;
                        break;
                    case "*":
                        value = num1 * num2;
                        break;
                }
                return value;
            }
        }
        String pattern = "^(\\d+)([+*-])\\(([\\d()*+-]+)\\)$";
        Pattern ptr = Pattern.compile(pattern);
        Matcher match = ptr.matcher(expression);
        if (match.find()) {
            Long number = Long.parseLong(match.group(1));
            String operand = match.group(2);
            String subexpression = match.group(3);
            switch (operand) {
                case "+":
                    return (number + CalculateExpression(subexpression));
                case "-":
                    return (number - CalculateExpression(subexpression));
                case "*":
                    return (number * CalculateExpression(subexpression));
            }
        }
        return 0l;
    }
}
